src/tools/compiletest/src/header.rs

```diff
@@ -574,12 +574,14 @@ impl Config {
         if line.starts_with(prefix) && line.as_bytes().get(prefix.len()) == Some(&b'-') {
             let name = line[prefix.len()+1 ..].split(&[':', ' '][..]).next().unwrap();
 
+            let triple: Vec<&str> = self.target.split('-').collect();
+
             name == "test" ||
-                util::matches_os(&self.target, name) ||             // target
-                name == util::get_arch(&self.target) ||             // architecture
-                name == util::get_pointer_width(&self.target) ||    // pointer width
-                name == self.stage_id.split('-').next().unwrap() || // stage
-                Some(name) == util::get_env(&self.target) ||        // env
+                util::matches_os(&self.target, &triple, name) || // target
```

Comment:
Nit: I think now that these functions look at the whole triple again, there's not much to be gained from splitting here and passing the component explicitly (especially for `matches_os` which needs the whole triple anyway). Perhaps just revert these changes and split in the `util::*` functions to keep the interface simpler?